spark/src/main/scala/org/apache/druid/spark/configuration/Configuration.scala

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.spark.configuration
+
+import org.apache.druid.java.util.common.StringUtils
+import org.apache.spark.sql.sources.v2.DataSourceOptions
+
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+
+/**
+  * A simple wrapper around a properties map that can also "dive" into a sub-Configuration if keys are stored using
+  * dotted separators. Because this class is designed to make working with Spark DataSourceOptions easier and DSOs are
+  * case-insensitive, this class is also case-insensitive.
+  *
+  * @param properties A Map of String property names to String property values to back this Configuration.
+  */
+class Configuration(properties: Map[String, String]) extends Serializable {

```

Comment:
Perhaps the reason of using this class that you mentioned in the PR description would be worth to add here.

Reply:
To me this feels like something that should be documented in the module documentation or in the coming DruidDataSourceV2 class (where the actual API will be exposed). If you think it makes more sense to include the reasoning with the Configuration class instead I'm happy to add it.

Reply:
Ah, I'm OK as long as it is documented somewhere. Thanks.